pallets/anchors-v2/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+// This file is part of Centrifuge chain project.
+
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::pallet_prelude::*;
+use frame_support::traits::ReservableCurrency;
+use frame_system::pallet_prelude::*;
+use scale_info::TypeInfo;
+use sp_runtime::{traits::AtLeast32BitUnsigned, FixedPointOperand};
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+pub mod weights;
+
+pub use pallet::*;
+pub use weights::*;
+
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub struct Anchor<AccountId, DocumentId, DocumentVersion, Hash, Balance> {
+	account_id: AccountId,
+	document_id: DocumentId,
+	document_version: DocumentVersion,
+	hash: Hash,
+	deposit: Balance,
+}
+
+pub type AnchorOf<T> = Anchor<
+	<T as frame_system::Config>::AccountId,
+	<T as Config>::DocumentId,
+	<T as Config>::DocumentVersion,
+	<T as frame_system::Config>::Hash,
+	<T as Config>::Balance,
+>;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		type Balance: Member
+			+ Parameter
+			+ AtLeast32BitUnsigned
+			+ Default
+			+ Copy
+			+ MaxEncodedLen
+			+ FixedPointOperand
+			+ From<u64>
+			+ From<u128>
+			+ TypeInfo
+			+ TryInto<u64>;
+
+		type Currency: ReservableCurrency<Self::AccountId, Balance = Self::Balance>;
+
+		/// Default deposit that will be taken when adding an anchor.
+		type DefaultAnchorDeposit: Get<Self::Balance>;
+
+		/// The type used to identify a document.
+		type DocumentId: Member
+			+ Parameter
+			+ AtLeast32BitUnsigned
+			+ Default
+			+ Copy
+			+ MaxEncodedLen
+			+ FixedPointOperand
+			+ From<u64>
+			+ From<u128>
```

Comment:
Does it really need to be a number?

Reply:
Yes, given the current spec that we have, the id should be a u128 and the version a u64.

Reply:
But from the pallet perspective, is it needed? Or is it something like we want to configure later the id as `u128` and the version as `u64` in the runtime?

Reply:
Hmm, I think that we'll always use `u128`/`u64`.

Reply:
Then, do we need associated types?

Reply:
Nope, I removed these.